gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RefreshUtil.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.google.common.base.Suppliers;
+import com.intellij.codeInsight.completion.BaseCompletionService;
+import com.intellij.codeInsight.completion.CompletionProcess;
+import com.intellij.codeInsight.completion.CompletionProgressIndicator;
+import com.intellij.codeInsight.completion.CompletionService;
+import com.intellij.openapi.application.ApplicationManager;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RefreshUtil {
+    private static final Logger log = LoggerFactory.getLogger(RefreshUtil.class);
+
+    public static Supplier<Void> refreshOnceSupplier() {
+        return Suppliers.memoize(() -> {
+            triggerRefresh();
+            return null;
+        });
+    }
+
+    private static void triggerRefresh() {
+        ApplicationManager.getApplication().invokeLater(() -> {
+            CompletionService completionService = CompletionService.getCompletionService();
+            if (completionService == null) {
+                return;
```

Comment:
I would throw something like an `IllegalStateException` here. Otherwise if this changes/breaks then the plugin just silently stops working. We want lots of errors so it shows up in logs etc and we can work out what's gone wrong (and people give us useful bug reports).

Reply:
Actually, for this one maybe it makes sense if this goes away when completion ends?? But definitely if we expect this to always exist.

Reply:
I'm pretty sure this should always exist